core/trino-main/src/main/java/io/trino/sql/planner/QueryPlanner.java

```diff
@@ -794,10 +798,12 @@ public MergeWriterNode plan(Merge merge)
             }
 
             ImmutableList.Builder<Expression> rowBuilder = ImmutableList.builder();
+            Assignments.Builder checkConstraintsAssignments = Assignments.builder();
```

Comment:
Calling this `checkConstraintAssignments` is confusing. It's *all* the assignments needed before the check constraints are evaluated, including those that are needed by downstream operations. Call it just `assignments`.